modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/utils/LogoSelector.js

```diff
@@ -0,0 +1,173 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import {sub} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+function LogoSelector({
+	defaultIcon,
+	disabled,
+	label,
+	logoId: initialLogoId,
+	logoURL: initialLogoURL,
+	namespace,
+	onChange,
+	selectLogoURL,
+}) {
+	const [values, setValues] = useState({
+		deleteLogo: false,
+		fileEntryId: initialLogoId || 0,
+		logoName: !initialLogoId
+			? Liferay.Language.get('default')
+			: sub(Liferay.Language.get('custom-x'), label),
+		logoURL: initialLogoURL,
+	});
+
+	const onChangeLogo = () => {
+		Liferay.Util.openWindow({
+			cache: false,
+			dialog: {
+				destroyOnHide: true,
+			},
+			dialogIframe: {
+				bodyCssClass: 'dialog-with-footer',
+			},
+			id: `${namespace}changeLogo`,
+			title: sub(Liferay.Language.get('upload-x'), label),
+			uri: selectLogoURL.replace(
+				escape('[$CURRENT_LOGO_URL$]'),
+				escape(logoURL)
+			),
+		});
+	};
+
+	const onClearLogo = () => {
+		setValues({
+			deleteLogo: true,
+			fileEntryId: 0,
+			logoName: Liferay.Language.get('default'),
+			logoURL: initialLogoURL,
+		});
+	};
+
+	useEffect(() => {
+		onChange({
+			target: {
+				name: 'logoId',
+				value: parseInt(values.fileEntryId, 10) || 0,
+			},
+		});
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [values]);
+
+	useEffect(() => {
+		const handleChangeLogo = ({
+			fileEntryId,
+			previewURL,
+			tempImageFileName,
+		}) => {
+			setValues({
+				deleteLogo: false,
+				fileEntryId,
+				logoName: tempImageFileName,
+				logoURL: previewURL,
+			});
+		};
+
+		Liferay.on('changeLogo', handleChangeLogo);
+
+		return () => {
+			Liferay.detach('changeLogo', handleChangeLogo);
+		};
+	}, []);
+
+	const {deleteLogo, fileEntryId, logoName, logoURL} = values;
+
+	return (
+		<>
+			<input
+				name={`${namespace}deleteLogo`}
+				type="hidden"
+				value={deleteLogo}
+			/>
+
+			<input
+				name={`${namespace}fileEntryId`}
+				type="hidden"
+				value={fileEntryId}
+			/>
+
+			{fileEntryId ? (
+				<img
+					alt={sub(Liferay.Language.get('current-x'), label)}
+					className="logo-selector-img mb-3"
+					src={logoURL}
+				/>
+			) : (
+				<svg className="logo-selector-default-img mb-3">
```

Comment:
ask confirmation to Censi

Reply:
Updated the account icon to the briefcase (also in the graph)